### DISCUSSION TITLE
How to import asset with dynamic url in public/asset folder?

### DISCUSSION THREAD

**Discussion by Alixchau** [remix-run/remix#6355] [Q&A]
I've been migrating an existing React project to Remix. In my react project I am able to import video cover photo from a dynamic url as blow:
const videoImg = process.env.PUBLIC_URL + `/video_thumbnail/${video?.img}`;

I'm able to access the video_thumbnail located under public folder. However Remix gives a process is no defined error when I use this line of code.

I've been searching for solutions but the they don't seem to work, any advise on this issue? Thank you.

**Comment by lili21** [comment]
you can't access `process.env` in component. check this one https://remix.run/docs/en/main/guides/envvars

**Reply by lili21** [reply]
you can read the `process.env` value inside loader and return it. 
Then you can read it in component by `useLoaderData`.

But I don't think you need do it in this way. you can just import the assets, and config the publicPath in remix.config.js and remix will handle the rest.
